### PR TITLE
[battery] Added hooks to expo battery

### PR DESCRIPTION
**Pull request by Adam-Schlichtmann**
# Why

Add some simple hooks to make usage of `expo-battery` easier.

# How

Use the existing listeners and types to create hooks.

# Test Plan

Import the hooks to the sandbox and use a simulator/emulator/device to verify changes to battery level, state, and power mode are reflected via the hook.

# Checklist

- [X] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [X] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [X] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Review by tsapeta — CHANGES_REQUESTED** [review]
Thanks for contribution! I left some suggestions regarding the doc comments. Also cc @amandeepmittal to review them.

Besides that, as far as I understand, all these hooks react only on some events, but what about updating the initial state to the current one? In other words, shouldn't `useBatteryLevel` also call `getBatteryLevelAsync` initially to override the initial `-1` value?

File: packages/expo-battery/src/Battery.ts [hunk 1]
```diff
@@ -182,6 +183,98 @@ export function addLowPowerModeListener(listener: (event: PowerModeEvent) => voi
...
+// @needsAudit
+/**
+ * ```ts
+ * const batteryLevel = useBatteryLevel();
+ * ```
+ * Returns the battery level of the device
+ *
+ * @see addBatteryLevelListener
+ * @return The battery level of the device
+ */
```
[the remark this answers is not in the document]
Comment (amandeepmittal) [reply packages/expo-battery/src/Battery.ts:195]: We don't usually explicitly call that a hook is a "React hook". I'd suggest excluding that context from all these definitions.

Also, I'd suggest using a simpler phrase like and call things for what they are for. "updates with the battery..." without explicitly mentioning React hook will be confusing.

```markdown
Gets the device's battery level, as in [`getBatteryLevelAsync`](#getbatterylevelasync).
```

File: packages/expo-battery/src/Battery.ts [hunk 2]
```diff
@@ -182,6 +183,98 @@ export function addLowPowerModeListener(listener: (event: PowerModeEvent) => voi
...
+// @needsAudit
+/**
+ * ```ts
+ * const batteryState = useBatteryState();
+ * ```
+ * Returns the battery state of the device
+ *
+ * @see addBatteryStateListener
+ * @return The battery state of the device
+ */
```
[the remark this answers is not in the document]
Comment (amandeepmittal) [reply packages/expo-battery/src/Battery.ts:216]:
```markdown
Gets the device's battery state, as in [`getBatteryStateAsync`](#getbatterystateasync).
```

File: packages/expo-battery/src/Battery.ts [hunk 3]
```diff
@@ -182,6 +183,98 @@ export function addLowPowerModeListener(listener: (event: PowerModeEvent) => voi
...
+// @needsAudit
+/**
+ * ```ts
+ * const lowPowerMode = useLowPowerMode();
+ * ```
+ * Returns boolean indicating if the device is in low power mode
+ *
+ * @see addLowPowerModeListener
+ * @return boolean indicating if the device is in low power mode
+ */
```
[the remark this answers is not in the document]
Comment (amandeepmittal) [reply packages/expo-battery/src/Battery.ts:237]:
```markdown
Boolean that indicates if the device is in low power or power saver mode, as in  [`isLowPowerModeEnabledAsync`](#islowpowermodeenabledasync).
```

File: packages/expo-battery/src/Battery.ts [hunk 4]
```diff
@@ -182,6 +183,98 @@ export function addLowPowerModeListener(listener: (event: PowerModeEvent) => voi
...
+// @needsAudit
+/**
+ * ```ts
+ * const { lowPowerMode, batteryLevel, batteryState } = usePowerState();
+ * ```
+ * Returns power state information
+ *
+ * @see PowerState
+ * @return power state information
+ */
```
[the remark this answers is not in the document]
Comment (amandeepmittal) [reply packages/expo-battery/src/Battery.ts:258]:
```markdown
Gets the device's power state information, as in [`getPowerStateAsync`](#getpowerstateasync).
```

**Review by amandeepmittal — COMMENTED** [review]
Left comments to clarify each hook's definition.

**Comment by Adam-Schlichtmann** [comment]
Thanks for the reviews @tsapeta and @amandeepmittal!

I believe everything should be updated as you requested.

**Review by amandeepmittal — APPROVED** [review]
Thank you for updating the type definitions 👍

**Review by tsapeta — APPROVED** [review]
Great, looks good to me now! Thanks for a good contribution 🤗 
I've just approved the CI jobs to run, I'll merge it later when they pass 😉

**Pull request merged**